staging/src/k8s.io/apiextensions-apiserver/examples/client-go/pkg/client/informers/externalversions/factory.go

```diff
@@ -91,12 +118,14 @@ func NewFilteredSharedInformerFactory(client versioned.Interface, defaultResync
 // NewSharedInformerFactoryWithOptions constructs a new instance of a SharedInformerFactory with additional options.
 func NewSharedInformerFactoryWithOptions(client versioned.Interface, defaultResync time.Duration, options ...SharedInformerOption) SharedInformerFactory {
 	factory := &sharedInformerFactory{
-		client:           client,
-		namespace:        v1.NamespaceAll,
-		defaultResync:    defaultResync,
-		informers:        make(map[reflect.Type]cache.SharedIndexInformer),
-		startedInformers: make(map[reflect.Type]bool),
-		customResync:     make(map[reflect.Type]time.Duration),
+		client:                       client,
+		namespace:                    v1.NamespaceAll,
+		defaultResync:                defaultResync,
+		informers:                    make(map[reflect.Type]cache.SharedIndexInformer),
+		startedInformers:             make(map[reflect.Type]bool),
+		customResync:                 make(map[reflect.Type]time.Duration),
+		extraClusterScopedIndexers:   cache.Indexers{},
+		extraNamespaceScopedIndexers: cache.Indexers{},
```

Comment:
why make the distinction between clusterScoped and namespaceScoped for indexers? Aren't they strictly defined by their key functions?